docs/DeveloperGuide.md

```diff
@@ -1,34 +1,511 @@
+
+The `Main` class is responsible for,
+* At app launch: Initializes the main components in the correct sequence and read data from our data files
+* At shut down: Terminates the continuous loop and shut down the components
+
+`Common` represents a collection of classes used by multiple other components.
+
+The rest of the App consists of four components.
+* `UserInterface`: The user interface of the App.
+* `Controller`: User input parser and command executor
+* `Model`: Holds the data of the App in memory
+* `Storage`: Reads data from, and writes data to, the hard disk.
+
+Each of the four components:
+The interaction between them is defined in an interface.
+Exposes its functionality using a concrete {Component Name}Manager class.
+
+The sections below give more details of each component.
+
+#### UserInterface component
+
+![userinterfacecomponent](https://raw.githubusercontent.com/AY2021S1-CS2113T-W12-4/tp/master/docs/diagram/UserInterfaceComponent.png)
+
+Figure 2. Class Diagram of the UserInterface component
+
+API: UserInterface.java
+
+The UserInterface component,
+* Prompts commands from the user.
+* Execute user commands using the Controller component.
+* Displays information based on changes to Model data.
+* Prints the user’s events of the week in a timetable format.
+
+This component uses the singleton design, meaning that there is only an instance required throughout the entire lifetime of the application, obtained with the following command
+`UserInterface userInterface = UserInterface.getInstance()`
+<br />
+<br />
+
+#### Controller component
+
+![controllercomponent](https://raw.githubusercontent.com/AY2021S1-CS2113T-W12-4/tp/master/docs/diagram/ControllerComponent.png)
+
+Figure 3. Class Diagram of the Controller component
+
+API: ControlManager.java
+
+The Controller component,
+* Receives user input from the user interface.
+* Extracts the command and model type from the user input.
+* Generate the required command from the CommandFactory.java class
+* Extract the required model to be accessed and modified.
+* Executes the command with the corresponding model.
+
+<br />
+<br />
+
+#### Model component
+
+![modelcomponent](https://raw.githubusercontent.com/AY2021S1-CS2113T-W12-4/tp/master/docs/diagram/ModelComponent.png)
+
+Figure 4. Class Diagram of the Model component
+
+API: Model.java
+
+The Model component,
+* Holds all the in-memory data of type event, quiz and contact.
+* Each data type has a corresponding manager that the controller can interface with. This is named as <datatype>Manager.java.
+* During a command execution, the manager will handle the modification and reading of its data type.
+* After the execution, the corresponding output is then passed on to the User Interface component to be shown to the user.
+
+<br />
+<br />
+
+#### Storage component
+![storagecomponent](https://raw.githubusercontent.com/AY2021S1-CS2113T-W12-4/tp/master/docs/diagram/StorageComponent.png)
```

Comment:
![image](https://user-images.githubusercontent.com/47527482/97396487-28dbad80-1922-11eb-9d5b-d76ff0e3f7a4.png)
Is StorageManager an interface? why are there dotted arrows pointing towards it